include/nuttx/semaphore.h

```diff
@@ -59,6 +59,54 @@
      {(c), (f), SEM_WAITLIST_INITIALIZER}
 #endif /* CONFIG_PRIORITY_INHERITANCE */
 
+/* Fast mutex lock/unlock path */
+
+#ifndef CONFIG_LIBC_ARCH_ATOMIC
```

Comment:
why need check CONFIG_LIBC_ARCH_ATOMIC

Reply:
@jlaitine please consider including this comment in the source code